pontoon/sync/formats/json_extensions.py

```diff
@@ -0,0 +1,138 @@
+"""
+Parser for the .json translation format as used by the WebExtensions API:
+https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Internationalization
+
+See also:
+https://www.chromium.org/developers/design-documents/extensions/how-the-extension-system-works/i18n
+"""
+import codecs
+import copy
+import json
+import logging
+import os
+
+from pontoon.sync import SyncError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+
+log = logging.getLogger(__name__)
+
+
+class JSONEntity(VCSTranslation):
+    """
+    Represents an entity in a JSON file.
+    """
+    def __init__(self, order, key, data):
+        self.key = key
+        self.data = data
+        self.order = order
+        self.strings = {None: self.source_string} if self.source_string else {}
+        self.last_translator = None
+        self.last_update = None
+
+    @property
+    def source_string(self):
+        return self.data['message']
+
+    @property
+    def source_string_plural(self):
+        return ''
+
+    @property
+    def comments(self):
+        return [self.data['description']] if 'description' in self.data else []
+
+    @property
+    def fuzzy(self):
+        return False
+
+    @fuzzy.setter
+    def fuzzy(self, fuzzy):
+        pass  # We don't use fuzzy in JSON
+
+    @property
+    def source(self):
+        return self.data['placeholders'] if 'placeholders' in self.data else []
+
+
+class JSONResource(ParsedResource):
+    def __init__(self, path, source_resource=None):
+        self.path = path
+        self.entities = {}
+        self.source_resource = source_resource
+
+        # Copy entities from the source_resource if it's available.
+        if source_resource:
+            for key, entity in source_resource.entities.items():
+                self.entities[key] = JSONEntity(
+                    entity.order,
+                    entity.key,
+                    copy.copy(entity.data),
+                )
+
+        try:
+            with codecs.open(path, 'r', 'utf-8') as resource:
+                content = resource.read()
+                self.json_file = json.loads(content)
```

Comment:
unless there is some setting somewhere else this will use dictionaries for json objects - ie unordered

you can set it globally (which it might be, but i cant see it grepping), or in the call to `loads` - eg

```
json.loads(content, object_pairs_hook=OrderedDict)
```